lib/sda/service.go

```diff
@@ -278,3 +282,120 @@ func CreateServiceMessage(service string, r interface{}) (*ServiceMessage, error
 	}, nil
 
 }
+
+/*
+A simple client structure to be used when wanting to connect to services. It
+holds the private and public key and allows to connect to a service through
+the network.
+The error-handling is done using the ErrorRet structure which can be returned
+in place of the standard reply. The Client.Send method will catch that and return
+ the appropriate error.
+*/
+
+// Client for a service
+type Client struct {
+	Private abstract.Secret
+	*network.Entity
+	ServiceID ServiceID
+}
+
+// NewClient returns a random client using the service s
+func NewClient(s string) *Client {
+	kp := config.NewKeyPair(network.Suite)
+	return &Client{
+		Entity:    network.NewEntity(kp.Public, ""),
+		Private:   kp.Secret,
+		ServiceID: ServiceFactory.ServiceID(s),
+	}
+}
+
+// NetworkSend opens the connection to 'dst' and sends the message 'req'. The
+// reply is returned, or an error if the timeout of 10 seconds is reached.
+func (c *Client) Send(dst *network.Entity, msg network.ProtocolMessage) (*network.Message, error) {
+	client := network.NewSecureTCPHost(c.Private, c.Entity)
+
+	// Connect to the root
+	dbg.Lvl4("Opening connection to", dst)
+	con, err := client.Open(dst)
+	defer client.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	m, err := network.NewNetworkMessage(msg)
+	if err != nil {
+		return nil, err
+	}
+	b, err := m.MarshalBinary()
+	if err != nil{
+		return nil, err
+	}
+	serviceReq := &ClientRequest{
+		Service: c.ServiceID,
+		Type:    m.MsgType,
+		Data:    b,
+	}
+	pchan := make(chan network.Message)
+	go func() {
+		// send the request
+		dbg.Lvlf3("Sending request %+v", serviceReq)
+		if err := con.Send(context.TODO(), serviceReq); err != nil {
+			close(pchan)
+			return
+		}
+		dbg.Lvl4("Waiting for the response")
+		// wait for the response
+		packet, err := con.Receive(context.TODO())
+		if err != nil {
+			close(pchan)
+			return
+		}
+		pchan <- packet
+	}()
+	select {
+	case response := <-pchan:
+		dbg.Lvlf5("Response: %+v", response)
+		// Catch an eventual error
+		err := ErrMsg(&response, nil)
+		if err != nil {
+			return nil, err
+		}
+		return &response, nil
+	case <-time.After(time.Second * 10):
```

Comment:
make it a field of `Client` so we don't ever have comments like "my simulation timeouted !" :D